chirp/drivers/baofeng_digital.py

```diff
@@ -0,0 +1,492 @@
+                mem.duplex = '-' if offset < 0 else '+'
+                mem.offset = abs(offset)
+            else:
+                mem.duplex = 'split'
+                mem.offset = tx_freq
+
+            if raw.digital:
+                mem.mode = raw.narrow and 'DN' or 'DIG'
+                mem.tmode = ''
+                mem.dtcs_polarity = 'NN'
+                encryption_key = raw.encryption_key
+            else:
+                mem.mode = raw.narrow and 'NFM' or 'FM'
+                hopping = raw.hopping
+
+                tx_tmode, tx_tvalue, tx_tpol = self._parse_tone(raw.tx_tone)
+                if tx_tmode == 'Tone':
+                    mem.rtone = tx_tvalue
+                elif tx_tmode == 'DTCS':
+                    mem.dtcs = tx_tvalue
+
+                rx_tmode, rx_tvalue, rx_tpol = self._parse_tone(raw.rx_tone)
+                if rx_tmode == 'Tone':
+                    mem.ctone = rx_tvalue
+                elif rx_tmode == 'DTCS':
+                    mem.rx_dtcs = rx_tvalue
+
+                if tx_tmode == 'Tone' and rx_tmode == '':
+                    mem.tmode = 'Tone'
+                elif (
+                    tx_tmode and
+                    tx_tmode == rx_tmode and
+                    tx_tvalue == rx_tvalue
+                ):
+                    mem.tmode = tx_tmode == 'Tone' and 'TSQL' or 'DTCS'
+                elif tx_tmode or rx_tmode:
+                    mem.tmode = 'Cross'
+                    mem.cross_mode = f'{tx_tmode}->{rx_tmode}'
+
+                mem.dtcs_polarity = tx_tpol + rx_tpol
+
+            mem.power = self.POWER_LEVELS[raw.high_power]
+            mem.skip = '' if raw.scan else 'S'
+            bcl = not raw.bcl_disabled
+
+        mem.extra = RadioSettingGroup("Extra", "extra")
+
+        rs = RadioSetting("bcl", "BCL",
+                          RadioSettingValueBoolean(bcl))
+        rs.set_doc('Busy channel lockout.')
+        mem.extra.append(rs)
+
+        rs = RadioSetting("hopping", "Hopping",
+                          RadioSettingValueBoolean(hopping))
+        rs.set_doc('Frequency hopping. Only supported on analog modes.')
+        mem.extra.append(rs)
+
+        rs = RadioSetting("encryption_key", "Encryption key",
+                          RadioSettingValueInteger(0, 31, encryption_key))
+        rs.set_doc('Encryption key. Only for digital modes.')
+        mem.extra.append(rs)
+
+        return mem
+
+    def _pack_ctcss(self, freq):
+        freq = int(freq * 10)
+        # Pack in BCD
+        return (
+            0x1000 * (freq // 1000 % 10) |
+            0x100 * (freq // 100 % 10) |
+            0x10 * (freq // 10 % 10) |
+            freq % 10
+        )
+
+    def _pack_dtcs(self, code, polarity):
+        return (
+            0x8000 |
+            (polarity == 'R' and 0x4000) |
+            0x1000 * (code // 1000 % 10) |
+            0x100 * (code // 100 % 10) |
+            0x10 * (code // 10 % 10) |
+            code % 10
+        )
```

Comment:
Same for these, surely you can re-use the bitwise functions and still or-in your flag right?